src/main/java/org/plp/implementations/plpisa/PlpFile.java

```diff
@@ -0,0 +1,132 @@
+package org.plp.implementations.plpisa;
+
+import org.plp.isa.AsmFile;
+import org.plp.isa.AsmProgram;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This is an in memory representation of an asm file {@link AsmFile}.
+ * This will map to a file path and will be part of {@link PlpProgram}
+ */
+public class PlpFile implements AsmFile {
+    private final String fileName;
```

Comment:
This will map to a file **path** and will be part of {@link PlpProgram}.

In the {@link PlpProgram} we are storing the prefix of the directory as Path object.

So, filePath will be a better name? Then in the constructor, we concatenate "fileName" + "directory" to be a "path". (final variable can't change the value after all).

Reply:
Yes updated to filePath